api/host_query_xjoin.py

```diff
@@ -179,6 +186,13 @@ def build_filter(field_name, field_value, field_type, operation, filter_building
         return filter_building_function(field_name, field_value[operation])
 
 
+def build_filter_string_multiple(field_name, field_value, field_type, operation):
+    if isinstance(field_value, field_type) or isinstance(field_value[operation], field_type):
+        return build_filter(field_name, field_value, str, operation, _nullable_string_filter)
+    else:
```

Comment:
Since we have basically no validation of these from the api spec (and this recently bit Ronny) I'd like to at least get some decent validation coverage in the executive code for new filters. you don't have to go add checks to everything in the file, but right here could use one. Instead of assuming anything that doesn't fit the first case is a list, an explicit check that it's a list and an else that raises a validation exception would be nice.
```suggestion
    if isinstance(field_value, list) or isinstance(field_value[operation], list):
        return build_filter(field_name, field_value[operation], list, operation, _nullable_multiple_string_filter)
    else:
        raise ValidationException(f"field_name expected to be a string or array of strings, instead got {type(field_value)}")
```
something like that. The suggestion is not precise.

Reply:
Added a `ValidationException` and added tests for it as well :+1: